<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Helping us debug nanopolish &mdash; pyclustermap 1.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> pyclustermap
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">pyclustermap</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Helping us debug nanopolish</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/debug.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="helping-us-debug-nanopolish">
<span id="help-us-debug"></span><h1>Helping us debug nanopolish<a class="headerlink" href="#helping-us-debug-nanopolish" title="Permalink to this headline"></a></h1>
<section id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Permalink to this headline"></a></h2>
<p>Running into errors with nanopolish? To help us debug, we need to be able to reproduce the errors. We can do this by packaging a subset of the files that were used by a nanopolish. We have provided <code class="docutils literal notranslate"><span class="pre">scripts/extract_reads_aligned_to_region.py</span></code> and this tutorial to help you do exactly this.</p>
<p>Briefly, this script will:</p>
<ul class="simple">
<li><p>extract reads that align to a given region in the draft genome assembly</p></li>
<li><p>rewrite a new BAM, BAI, FASTA file with these reads</p></li>
<li><p>extract the FAST5 files associated with these reads</p></li>
<li><p>save all these files into a tar.gz file</p></li>
</ul>
</section>
<section id="workflow">
<h2>Workflow<a class="headerlink" href="#workflow" title="Permalink to this headline"></a></h2>
<ol class="arabic simple">
<li><p>Narrow down a problematic region by running <code class="docutils literal notranslate"><span class="pre">nanopolish</span> <span class="pre">variants</span> <span class="pre">--consensus</span> <span class="pre">[...]</span></code> and changing the <code class="docutils literal notranslate"><span class="pre">-w</span></code> parameter.</p></li>
<li><p>Run the <code class="docutils literal notranslate"><span class="pre">scripts/extract_reads_aligned_to_region.py</span></code>.</p></li>
<li><p>Send the resulting <code class="docutils literal notranslate"><span class="pre">tar.gz</span></code> file to us by hosting either a dropbox or google drive.</p></li>
</ol>
</section>
<section id="tutorial-using-extraction-helper-script-to-create-example-datsets">
<span id="creating-example-dataset"></span><h2>Tutorial - using extraction helper script to create example datsets<a class="headerlink" href="#tutorial-using-extraction-helper-script-to-create-example-datsets" title="Permalink to this headline"></a></h2>
<p>We extracted a subset of reads for a 2kb region to create the example dataset for the eventalign and consensus tutorial using <code class="docutils literal notranslate"><span class="pre">scripts/extract_reads_aligned_to_region.py</span></code>. Here is how:</p>
<div class="line-block">
<div class="line"><br /></div>
</div>
<p>Generated basecalled <code class="docutils literal notranslate"><span class="pre">--reads</span></code> file:</p>
<ol class="arabic">
<li><p>Basecalled reads with albacore:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">read_fast5_basecaller</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">c</span> <span class="n">r94_450bps_linear</span><span class="o">.</span><span class="n">cfg</span> <span class="o">-</span><span class="n">t</span> <span class="mi">8</span> <span class="o">-</span><span class="n">i</span> <span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">raw</span><span class="o">/</span><span class="n">fast5</span><span class="o">/</span><span class="n">files</span> <span class="o">-</span><span class="n">s</span> <span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">albacore</span><span class="o">-</span><span class="mf">2.0.1</span><span class="o">/</span><span class="n">output</span><span class="o">/</span><span class="n">directory</span> <span class="o">-</span><span class="n">o</span> <span class="n">fastq</span>
</pre></div>
</div>
</li>
<li><p>Merged the different albacore fastq outputs:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cat</span> <span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">albacore</span><span class="o">-</span><span class="mf">2.0.1</span><span class="o">/</span><span class="n">output</span><span class="o">/</span><span class="n">directory</span><span class="o">/</span><span class="n">workspace</span><span class="o">/</span><span class="k">pass</span><span class="o">/*.</span><span class="n">fastq</span> <span class="o">&gt;</span> <span class="n">albacore</span><span class="o">-</span><span class="mf">2.0.1</span><span class="o">-</span><span class="n">merged</span><span class="o">.</span><span class="n">fastq</span>
</pre></div>
</div>
</li>
<li><p>Converted the merged fastq to fasta format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">paste</span> <span class="o">-</span> <span class="o">-</span> <span class="o">-</span> <span class="o">-</span> <span class="o">&lt;</span> <span class="n">albacore</span><span class="o">-</span><span class="mf">2.0.1</span><span class="o">-</span><span class="n">merged</span><span class="o">.</span><span class="n">fastq</span> <span class="o">|</span> <span class="n">cut</span> <span class="o">-</span><span class="n">f</span> <span class="mi">1</span><span class="p">,</span><span class="mi">2</span> <span class="o">|</span> <span class="n">sed</span> <span class="s1">&#39;s/^@/&gt;/&#39;</span> <span class="o">|</span> <span class="n">tr</span> <span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span> <span class="s2">&quot;</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">&gt;</span> <span class="n">reads</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
</li>
</ol>
<div class="line-block">
<div class="line"><br /></div>
</div>
<p>Generated <code class="docutils literal notranslate"><span class="pre">--bam</span></code> file with the draft genome assembly (<code class="docutils literal notranslate"><span class="pre">-g</span></code>):</p>
<ol class="arabic">
<li><p>Ran canu to create draft genome assembly:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">canu</span> \
    <span class="o">-</span><span class="n">p</span> <span class="n">ecoli</span> <span class="o">-</span><span class="n">d</span> <span class="n">outdir</span> <span class="n">genomeSize</span><span class="o">=</span><span class="mf">4.6</span><span class="n">m</span> \
    <span class="o">-</span><span class="n">nanopore</span><span class="o">-</span><span class="n">raw</span> <span class="n">reads</span><span class="o">.</span><span class="n">fasta</span> \
</pre></div>
</div>
</li>
<li><p>Index draft assembly:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bwa</span> <span class="n">index</span> <span class="n">ecoli</span><span class="o">.</span><span class="n">contigs</span><span class="o">.</span><span class="n">fasta</span>
<span class="n">samtools</span> <span class="n">faidx</span> <span class="n">ecoli</span><span class="o">.</span><span class="n">contigs</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
</li>
<li><p>Aligned reads to draft genome assembly with bwa (v0.7.12):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bwa</span> <span class="n">mem</span> <span class="o">-</span><span class="n">x</span> <span class="n">ont2d</span> <span class="n">ecoli</span><span class="o">.</span><span class="n">contigs</span><span class="o">.</span><span class="n">fasta</span> <span class="n">reads</span><span class="o">.</span><span class="n">fasta</span> <span class="o">|</span> <span class="n">samtools</span> <span class="n">sort</span> <span class="o">-</span><span class="n">o</span> <span class="n">reads</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> <span class="o">-</span><span class="n">T</span> <span class="n">reads</span><span class="o">.</span><span class="n">tmp</span>
<span class="n">samtools</span> <span class="n">index</span> <span class="n">reads</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
</li>
</ol>
<div class="line-block">
<div class="line"><br /></div>
</div>
<p>Selected a <code class="docutils literal notranslate"><span class="pre">--window</span></code>:</p>
<ol class="arabic">
<li><p>Identified the first contig name and chose a random start position:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">head</span> <span class="o">-</span><span class="mi">3</span> <span class="n">ecoli</span><span class="o">.</span><span class="n">contigs</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
</li>
</ol>
<p>Output:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span><span class="n">tig00000001</span> <span class="nb">len</span><span class="o">=</span><span class="mi">4376233</span> <span class="n">reads</span><span class="o">=</span><span class="mi">23096</span> <span class="n">covStat</span><span class="o">=</span><span class="mf">7751.73</span> <span class="n">gappedBases</span><span class="o">=</span><span class="n">no</span> <span class="n">class</span><span class="o">=</span><span class="n">contig</span> <span class="n">suggestRepeat</span><span class="o">=</span><span class="n">no</span> <span class="n">suggestCircular</span><span class="o">=</span><span class="n">no</span>
<span class="n">AGATGCTTTGAAAGAAACGCAGAATAGATCTCTATGTAATGATATGGAATACTCTGGTATTGTCTGTAAAGATACTAATGGAAAATATTTTGCATCTAAG</span>
<span class="n">GCAGAAACTGATAATTTAAGAAAGGAGTCATATCCTCTGAAAAGAAAATGTCCCACAGGTACAGATAGAGTTGCTGCTTATCATACTCACGGTGCAGATA</span>
</pre></div>
</div>
<p>As we wanted a 2kb region, we selected a random start position (200000) so our end position was 202000. Therefore our <code class="docutils literal notranslate"><span class="pre">--window</span></code> was “tig00000001:200000-202000”.</p>
<div class="line-block">
<div class="line"><br /></div>
</div>
<p>Using the files we created, we ran <code class="docutils literal notranslate"><span class="pre">scripts/extract_reads_aligned_to_region.py</span></code>, please see usage example below.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Make sure nanopolish still reproduces the same error on this subset before sending it to us.</p>
</div>
</section>
<section id="usage-example">
<h2>Usage example<a class="headerlink" href="#usage-example" title="Permalink to this headline"></a></h2>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python3</span> <span class="n">extract_reads_aligned_to_region</span><span class="o">.</span><span class="n">py</span> \
    <span class="o">--</span><span class="n">reads</span> <span class="n">reads</span><span class="o">.</span><span class="n">fasta</span> \
    <span class="o">--</span><span class="n">genome</span> <span class="n">ecoli</span><span class="o">.</span><span class="n">contigs</span><span class="o">.</span><span class="n">fasta</span> \
    <span class="o">--</span><span class="n">bam</span> <span class="n">reads</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> \
    <span class="o">--</span><span class="n">window</span> <span class="s2">&quot;tig00000001:200000-202000&quot;</span> \
    <span class="o">--</span><span class="n">output_prefix</span> <span class="n">ecoli_2kb_region</span> <span class="o">--</span><span class="n">verbose</span>
</pre></div>
</div>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 25%" />
<col style="width: 5%" />
<col style="width: 20%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Argument name(s)</p></th>
<th class="head"><p>Req.</p></th>
<th class="head"><p>Default value</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-b</span></code>, <code class="docutils literal notranslate"><span class="pre">--bam</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>Sorted bam file created by aligning reads to the draft genome.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-g</span></code>, <code class="docutils literal notranslate"><span class="pre">--genome</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>Draft genome assembly</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>, <code class="docutils literal notranslate"><span class="pre">--reads</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>Fasta, fastq, fasta.gz, or fastq.gz file containing basecalled reads.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-w</span></code>, <code class="docutils literal notranslate"><span class="pre">--window</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>Draft genome assembly coordinate string ex. “contig:start-end”. It is essential that you wrap the coordinates in quotation marks (&quot;).</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>, <code class="docutils literal notranslate"><span class="pre">--output_prefix</span></code></p></td>
<td><p>N</p></td>
<td><p>reads_subset</p></td>
<td><p>Prefix of output tar.gz and log file.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-v</span></code>, <code class="docutils literal notranslate"><span class="pre">--verbose</span></code></p></td>
<td><p>N</p></td>
<td><p>False</p></td>
<td><p>Use for verbose output with info on progress.</p></td>
</tr>
</tbody>
</table>
</section>
<section id="script-overview">
<h2>Script overview<a class="headerlink" href="#script-overview" title="Permalink to this headline"></a></h2>
<ol class="arabic simple">
<li><p>Parse input files</p></li>
<li><p>Assumes readdb file name from input reads file</p></li>
<li><dl class="simple">
<dt>Validates input</dt><dd><ul class="simple">
<li><p>checks that input bam, readdb, fasta/q, draft genome assembly, draft genome assembly index file exist, are not empy, and are readable</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>With user input draft genome assembly coordinates, extracts all reads that aligned within these coordinates stores the read_ids. This information can be found from the input BAM.</dt><dd><ul class="simple">
<li><p>uses pysam.AlignmentFile</p></li>
<li><p>uses samfile.fetch(region=draft_ga_coords) to get all reads aligned to region</p></li>
<li><p>if reads map to multiple sections within draft ga it is not added again</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Parses through the input readdb file to find the FAST5 files associated with each region that aligned to region</dt><dd><ul class="simple">
<li><p>stores in dictionary region_fast5_files; key = read_id, value = path/to/fast5/file</p></li>
<li><p>path to fast5 file is currently dependent on the user’s directory structure</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Make a BAM and BAI file for this specific region</dt><dd><ul class="simple">
<li><p>creates a new BAM file called <code class="docutils literal notranslate"><span class="pre">region.bam</span></code></p></li>
<li><p>with pysam.view we rewrite the new bam with reads that aligned to the region…</p></li>
<li><p>with pysam.index we create a new BAI file</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Now to make a new FASTA file with this subset of reads</dt><dd><ul class="simple">
<li><p>the new fasta file is called <code class="docutils literal notranslate"><span class="pre">region.fasta</span></code></p></li>
<li><p>this first checks what type of sequences file is given { <code class="docutils literal notranslate"><span class="pre">fasta</span></code>, <code class="docutils literal notranslate"><span class="pre">fastq</span></code>, <code class="docutils literal notranslate"><span class="pre">fasta.gz</span></code>, <code class="docutils literal notranslate"><span class="pre">fastq.gz</span></code> }</p></li>
<li><p>then handles based on type of seq file using SeqIO.parse</p></li>
<li><p>then writes to a new fasta file</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Let’s get to tarring</dt><dd><ul class="simple">
<li><p>creates a <code class="docutils literal notranslate"><span class="pre">tar.gz</span></code> file with the output prefix</p></li>
<li><p>saves the fast5 files in directory <code class="docutils literal notranslate"><span class="pre">output_prefix/fast5_files/</span></code></p></li>
</ul>
</dd>
</dl>
</li>
<li><p>Adds the new fasta, new bam, and new bai file with the subset of reads</p></li>
<li><p>Adds the draft genome asssembly and associated fai index file</p></li>
<li><dl class="simple">
<dt>Performs a check</dt><dd><ul class="simple">
<li><p>the number of reads in the new BAM file, new FASTA file, and the number of files in the fast5_files directory should be equal</p></li>
</ul>
</dd>
</dl>
</li>
<li><p>Outputs a <code class="docutils literal notranslate"><span class="pre">tar.gz</span></code> and <code class="docutils literal notranslate"><span class="pre">log</span></code> file. FIN!</p></li>
</ol>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Wubin Ding.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>