<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Manual &mdash; pyclustermap 1.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> pyclustermap
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">pyclustermap</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Manual</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/manual.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="manual">
<span id="id1"></span><h1>Manual<a class="headerlink" href="#manual" title="Permalink to this headline"></a></h1>
<p>Modules available:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">nanopolish</span> <span class="n">extract</span><span class="p">:</span> <span class="n">extract</span> <span class="n">reads</span> <span class="ow">in</span> <span class="n">FASTA</span> <span class="ow">or</span> <span class="n">FASTQ</span> <span class="nb">format</span> <span class="kn">from</span> <span class="nn">a</span> <span class="n">directory</span> <span class="n">of</span> <span class="n">FAST5</span> <span class="n">files</span>
<span class="n">nanopolish</span> <span class="n">call</span><span class="o">-</span><span class="n">methylation</span><span class="p">:</span> <span class="n">predict</span> <span class="n">genomic</span> <span class="n">bases</span> <span class="n">that</span> <span class="n">may</span> <span class="n">be</span> <span class="n">methylated</span>
<span class="n">nanopolish</span> <span class="n">variants</span><span class="p">:</span> <span class="n">detect</span> <span class="n">SNPs</span> <span class="ow">and</span> <span class="n">indels</span> <span class="k">with</span> <span class="n">respect</span> <span class="n">to</span> <span class="n">a</span> <span class="n">reference</span> <span class="n">genome</span>
<span class="n">nanopolish</span> <span class="n">variants</span> <span class="o">--</span><span class="n">consensus</span><span class="p">:</span> <span class="n">calculate</span> <span class="n">an</span> <span class="n">improved</span> <span class="n">consensus</span> <span class="n">sequence</span> <span class="k">for</span> <span class="n">a</span> <span class="n">draft</span> <span class="n">genome</span> <span class="n">assembly</span>
<span class="n">nanopolish</span> <span class="n">eventalign</span><span class="p">:</span> <span class="n">align</span> <span class="n">signal</span><span class="o">-</span><span class="n">level</span> <span class="n">events</span> <span class="n">to</span> <span class="n">k</span><span class="o">-</span><span class="n">mers</span> <span class="n">of</span> <span class="n">a</span> <span class="n">reference</span> <span class="n">genome</span>
<span class="n">nanopolish</span> <span class="n">phase</span><span class="o">-</span><span class="n">reads</span><span class="p">:</span> <span class="n">Phase</span> <span class="n">reads</span> <span class="n">using</span> <span class="n">heterozygous</span> <span class="n">SNVs</span> <span class="k">with</span> <span class="n">respect</span> <span class="n">to</span> <span class="n">a</span> <span class="n">reference</span> <span class="n">genome</span>
<span class="n">nanopolish</span> <span class="n">polya</span><span class="p">:</span> <span class="n">Estimate</span> <span class="n">polyadenylated</span> <span class="n">tail</span> <span class="n">lengths</span> <span class="n">on</span> <span class="n">native</span> <span class="n">RNA</span> <span class="n">reads</span>
</pre></div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<section id="extract">
<h2>extract<a class="headerlink" href="#extract" title="Permalink to this headline"></a></h2>
<section id="overview">
<h3>Overview<a class="headerlink" href="#overview" title="Permalink to this headline"></a></h3>
<p>This module is used to extract reads in FASTA or FASTQ format from a directory of FAST5 files.</p>
</section>
<section id="input">
<h3>Input<a class="headerlink" href="#input" title="Permalink to this headline"></a></h3>
<blockquote>
<div><ul class="simple">
<li><p>path to a directory of FAST5 files modified to contain basecall information</p></li>
</ul>
</div></blockquote>
</section>
<section id="output">
<h3>Output<a class="headerlink" href="#output" title="Permalink to this headline"></a></h3>
<blockquote>
<div><ul class="simple">
<li><p>sequences of reads in FASTA or FASTQ format</p></li>
</ul>
</div></blockquote>
</section>
<section id="usage-example">
<h3>Usage example<a class="headerlink" href="#usage-example" title="Permalink to this headline"></a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">nanopolish</span> <span class="n">extract</span> <span class="p">[</span><span class="n">OPTIONS</span><span class="p">]</span> <span class="o">&lt;</span><span class="n">fast5</span><span class="o">|</span><span class="nb">dir</span><span class="o">&gt;</span>
</pre></div>
</div>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 10%" />
<col style="width: 20%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Argument name(s)</p></th>
<th class="head"><p>Required</p></th>
<th class="head"><p>Default value</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>&lt;fast5|dir&gt;</p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>FAST5 or path to directory of FAST5 files.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>, <code class="docutils literal notranslate"><span class="pre">--recurse</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>Recurse into subdirectories</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-q</span></code>, <code class="docutils literal notranslate"><span class="pre">--fastq</span></code></p></td>
<td><p>N</p></td>
<td><p>fasta format</p></td>
<td><p>Use when you want to extract to FASTQ format</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-t</span></code>, <code class="docutils literal notranslate"><span class="pre">--type=TYPE</span></code></p></td>
<td><p>N</p></td>
<td><p>2d-or-template</p></td>
<td><p>The type of read either: {template, complement, 2d, 2d-or-template, any}</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-b</span></code>, <code class="docutils literal notranslate"><span class="pre">--basecaller=NAME[:VERSION]</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>consider only data produced by basecaller NAME, optionally with given exact VERSION</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>, <code class="docutils literal notranslate"><span class="pre">--output=FILE</span></code></p></td>
<td><p>N</p></td>
<td><p>stdout</p></td>
<td><p>Write output to FILE</p></td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line"><br /></div>
</div>
</section>
</section>
<section id="index">
<h2>index<a class="headerlink" href="#index" title="Permalink to this headline"></a></h2>
<section id="id2">
<h3>Overview<a class="headerlink" href="#id2" title="Permalink to this headline"></a></h3>
<p>Build an index mapping from basecalled reads to the signals measured by the sequencer</p>
</section>
<section id="id3">
<h3>Input<a class="headerlink" href="#id3" title="Permalink to this headline"></a></h3>
<blockquote>
<div><ul class="simple">
<li><p>path to directory of raw nanopore sequencing data in FAST5 format</p></li>
<li><p>basecalled reads</p></li>
</ul>
</div></blockquote>
</section>
<section id="id4">
<h3>Output<a class="headerlink" href="#id4" title="Permalink to this headline"></a></h3>
<blockquote>
<div><ul class="simple">
<li><p>gzipped FASTA file of basecalled reads (.index)</p></li>
<li><p>index files (.fai, .gzi, .readdb)</p></li>
</ul>
</div></blockquote>
</section>
<section id="readdb-file-format">
<h3>Readdb file format<a class="headerlink" href="#readdb-file-format" title="Permalink to this headline"></a></h3>
<p>Readdb file is a tab-separated file that contains two columns. One column represents read ids and the other column represents the corresponding path to FAST5 file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">read_id_1</span>   <span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">fast5</span><span class="o">/</span><span class="n">containing</span><span class="o">/</span><span class="n">reads_id_1</span><span class="o">/</span><span class="n">signals</span>
<span class="n">read_id_2</span>   <span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">fast5</span><span class="o">/</span><span class="n">containing</span><span class="o">/</span><span class="n">read_id_2</span><span class="o">/</span><span class="n">signals</span>
</pre></div>
</div>
</section>
<section id="id5">
<h3>Usage example<a class="headerlink" href="#id5" title="Permalink to this headline"></a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">nanopolish</span> <span class="n">index</span> <span class="p">[</span><span class="n">OPTIONS</span><span class="p">]</span> <span class="o">-</span><span class="n">d</span> <span class="n">nanopore_raw_file_directory</span> <span class="n">reads</span><span class="o">.</span><span class="n">fastq</span>
</pre></div>
</div>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 10%" />
<col style="width: 20%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Argument name(s)</p></th>
<th class="head"><p>Required</p></th>
<th class="head"><p>Default value</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-d</span></code>, <code class="docutils literal notranslate"><span class="pre">--directory</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>FAST5 or path to directory of FAST5 files containing ONT sequencing raw signal information.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-f</span></code>, <code class="docutils literal notranslate"><span class="pre">--fast5-fofn</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>file containing the paths to each fast5 for the run</p></td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="call-methylation">
<h2>call-methylation<a class="headerlink" href="#call-methylation" title="Permalink to this headline"></a></h2>
<section id="id6">
<h3>Overview<a class="headerlink" href="#id6" title="Permalink to this headline"></a></h3>
<p>Classify nucleotides as methylated or not.</p>
</section>
<section id="id7">
<h3>Input<a class="headerlink" href="#id7" title="Permalink to this headline"></a></h3>
<blockquote>
<div><ul class="simple">
<li><p>Basecalled ONT reads in FASTA format</p></li>
</ul>
</div></blockquote>
</section>
<section id="id8">
<h3>Output<a class="headerlink" href="#id8" title="Permalink to this headline"></a></h3>
<blockquote>
<div><ul class="simple">
<li><p>tab-separated file containing per-read log-likelihood ratios</p></li>
</ul>
</div></blockquote>
</section>
<section id="id9">
<h3>Usage example<a class="headerlink" href="#id9" title="Permalink to this headline"></a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">nanopolish</span> <span class="n">call</span><span class="o">-</span><span class="n">methylation</span> <span class="p">[</span><span class="n">OPTIONS</span><span class="p">]</span> <span class="o">&lt;</span><span class="n">fast5</span><span class="o">|</span><span class="nb">dir</span><span class="o">&gt;</span>
</pre></div>
</div>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 10%" />
<col style="width: 20%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Argument name(s)</p></th>
<th class="head"><p>Required</p></th>
<th class="head"><p>Default value</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>, <code class="docutils literal notranslate"><span class="pre">--reads=FILE</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>the ONT reads are in fasta FILE</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-b</span></code>, <code class="docutils literal notranslate"><span class="pre">--bam=FILE</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>the reads aligned to the genome assembly are in bam FILE</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-g</span></code>, <code class="docutils literal notranslate"><span class="pre">--genome=FILE</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>the genome we are computing a consensus for is in FILE</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-t</span></code>, <code class="docutils literal notranslate"><span class="pre">--threads=NUM</span></code></p></td>
<td><p>N</p></td>
<td><p>1</p></td>
<td><p>use NUM threads</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--progress</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>print out a progress message</p></td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="variants">
<h2>variants<a class="headerlink" href="#variants" title="Permalink to this headline"></a></h2>
<section id="id10">
<h3>Overview<a class="headerlink" href="#id10" title="Permalink to this headline"></a></h3>
<p>This module is used to call single nucleotide polymorphisms (SNPs) using a signal-level HMM.</p>
</section>
<section id="id11">
<h3>Input<a class="headerlink" href="#id11" title="Permalink to this headline"></a></h3>
<blockquote>
<div><ul class="simple">
<li><p>basecalled reads</p></li>
<li><p>alignment info</p></li>
<li><p>genome assembly</p></li>
</ul>
</div></blockquote>
</section>
<section id="id12">
<h3>Output<a class="headerlink" href="#id12" title="Permalink to this headline"></a></h3>
<blockquote>
<div><ul class="simple">
<li><p>VCF file</p></li>
</ul>
</div></blockquote>
</section>
<section id="id13">
<h3>Usage example<a class="headerlink" href="#id13" title="Permalink to this headline"></a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">nanopolish</span> <span class="n">variants</span> <span class="p">[</span><span class="n">OPTIONS</span><span class="p">]</span> <span class="o">--</span><span class="n">reads</span> <span class="n">reads</span><span class="o">.</span><span class="n">fa</span> <span class="o">--</span><span class="n">bam</span> <span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">--</span><span class="n">genome</span> <span class="n">genome</span><span class="o">.</span><span class="n">fa</span>
</pre></div>
</div>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 10%" />
<col style="width: 20%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Argument name(s)</p></th>
<th class="head"><p>Required</p></th>
<th class="head"><p>Default value</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--snps</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>use flag to only call SNPs</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--consensus=FILE</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>run in consensus calling mode and write polished sequence to FILE</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--fix-homopolymers</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>use flag to run the experimental homopolymer caller</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--faster</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>minimize compute time while slightly reducing consensus accuracy</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-w</span></code>, <code class="docutils literal notranslate"><span class="pre">--window=STR</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>find variants in window STR (format: &lt;chromsome_name&gt;:&lt;start&gt;-&lt;end&gt;)</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>, <code class="docutils literal notranslate"><span class="pre">--reads=FILE</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>the ONT reads are in fasta FILE</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-b</span></code>, <code class="docutils literal notranslate"><span class="pre">--bam=FILE</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>the reads aligned to the reference genome are in bam FILE</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-e</span></code>, <code class="docutils literal notranslate"><span class="pre">--event-bam=FILE</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>the events aligned to the reference genome are in bam FILE</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-g</span></code>, <code class="docutils literal notranslate"><span class="pre">--genome=FILE</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>the reference genome is in FILE</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>, <code class="docutils literal notranslate"><span class="pre">--outfile=FILE</span></code></p></td>
<td><p>N</p></td>
<td><p>stdout</p></td>
<td><p>write result to FILE</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-t</span></code>, <code class="docutils literal notranslate"><span class="pre">--threads=NUM</span></code></p></td>
<td><p>N</p></td>
<td><p>1</p></td>
<td><p>use NUM threads</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-m</span></code>, <code class="docutils literal notranslate"><span class="pre">--min-candidate-frequency=F</span></code></p></td>
<td><p>N</p></td>
<td><p>0.2</p></td>
<td><p>extract candidate variants from the aligned reads when the variant frequency is at least F</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-d</span></code>, <code class="docutils literal notranslate"><span class="pre">--min-candidate-depth=D</span></code></p></td>
<td><p>N</p></td>
<td><p>20</p></td>
<td><p>extract candidate variants from the aligned reads when the depth is at least D</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>, <code class="docutils literal notranslate"><span class="pre">--max-haplotypes=N</span></code></p></td>
<td><p>N</p></td>
<td><p>1000</p></td>
<td><p>consider at most N haplotypes combinations</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--max-rounds=N</span></code></p></td>
<td><p>N</p></td>
<td><p>50</p></td>
<td><p>perform N rounds of consensus sequence improvement</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>, <code class="docutils literal notranslate"><span class="pre">--candidates=VCF</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>read variants candidates from VCF, rather than discovering them from aligned reads</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-a</span></code>, <code class="docutils literal notranslate"><span class="pre">--alternative-basecalls-bam=FILE</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>if an alternative basecaller was used that does not output event annotations then use basecalled sequences from FILE. The signal-level events will still be taken from the -b bam</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--calculate-all-support</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>when making a call, also calculate the support of the 3 other possible bases</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--models-fofn=FILE</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>read alternatives k-mer models from FILE</p></td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="event-align">
<h2>event align<a class="headerlink" href="#event-align" title="Permalink to this headline"></a></h2>
<section id="id14">
<h3>Overview<a class="headerlink" href="#id14" title="Permalink to this headline"></a></h3>
<p>Align nanopore events to reference k-mers</p>
</section>
<section id="id15">
<h3>Input<a class="headerlink" href="#id15" title="Permalink to this headline"></a></h3>
<blockquote>
<div><ul class="simple">
<li><p>basecalled reads</p></li>
<li><p>alignment information</p></li>
<li><p>assembled genome</p></li>
</ul>
</div></blockquote>
</section>
<section id="id16">
<h3>Usage example<a class="headerlink" href="#id16" title="Permalink to this headline"></a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">nanopolish</span> <span class="n">eventalign</span> <span class="p">[</span><span class="n">OPTIONS</span><span class="p">]</span> <span class="o">--</span><span class="n">reads</span> <span class="n">reads</span><span class="o">.</span><span class="n">fa</span> <span class="o">--</span><span class="n">bam</span> <span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">--</span><span class="n">genome</span> <span class="n">genome</span><span class="o">.</span><span class="n">fa</span>
</pre></div>
</div>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 10%" />
<col style="width: 20%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Argument name(s)</p></th>
<th class="head"><p>Required</p></th>
<th class="head"><p>Default value</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--sam</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>use to write output in SAM format</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-w,</span> <span class="pre">--window=STR</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>Compute the consensus for window STR (format : ctg:start_id-end_id)</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-r,</span> <span class="pre">--reads=FILE</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>the ONT reads are in fasta FILE</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-b,</span> <span class="pre">--bam=FILE</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>the reads aligned to the genome assembly are in bam FILE</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-g,</span> <span class="pre">--genome=FILE</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>the genome we are computing a consensus for is in FILE</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-t,</span> <span class="pre">--threads=NUM</span></code></p></td>
<td><p>N</p></td>
<td><p>1</p></td>
<td><p>use NUM threads</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--scale-events</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>scale events to the model, rather than vice-versa</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--progress</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>print out a progress message</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-n</span></code>, <code class="docutils literal notranslate"><span class="pre">--print-read-names</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>print read names instead of indexes</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--summary=FILE</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>summarize the alignment of each read/strand in FILE</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--samples</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>write the raw samples for the event to the tsv output</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--models-fofn=FILE</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>read alternative k-mer models from FILE</p></td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="phase-reads-experimental">
<h2>phase-reads - (experimental)<a class="headerlink" href="#phase-reads-experimental" title="Permalink to this headline"></a></h2>
<section id="id17">
<h3>Overview<a class="headerlink" href="#id17" title="Permalink to this headline"></a></h3>
<p>Phase reads using heterozygous SNVs with respect to a reference genome</p>
</section>
<section id="id18">
<h3>Input<a class="headerlink" href="#id18" title="Permalink to this headline"></a></h3>
<blockquote>
<div><ul class="simple">
<li><p>basecalled reads</p></li>
<li><p>alignment information</p></li>
<li><p>assembled genome</p></li>
<li><p>variants (from nanopolish variants or from other sources eg. Illumina VCF)</p></li>
</ul>
</div></blockquote>
</section>
<section id="id19">
<h3>Usage example<a class="headerlink" href="#id19" title="Permalink to this headline"></a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">nanopolish</span> <span class="n">phase</span><span class="o">-</span><span class="n">reads</span> <span class="p">[</span><span class="n">OPTIONS</span><span class="p">]</span> <span class="o">--</span><span class="n">reads</span> <span class="n">reads</span><span class="o">.</span><span class="n">fa</span> <span class="o">--</span><span class="n">bam</span> <span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">--</span><span class="n">genome</span> <span class="n">genome</span><span class="o">.</span><span class="n">fa</span> <span class="n">variants</span><span class="o">.</span><span class="n">vcf</span>
</pre></div>
</div>
</section>
</section>
<section id="polya">
<h2>polya<a class="headerlink" href="#polya" title="Permalink to this headline"></a></h2>
<section id="id20">
<h3>Overview<a class="headerlink" href="#id20" title="Permalink to this headline"></a></h3>
<p>Estimate the number of nucleotides in the poly(A) tails of native RNA reads.</p>
</section>
<section id="id21">
<h3>Input<a class="headerlink" href="#id21" title="Permalink to this headline"></a></h3>
<blockquote>
<div><ul class="simple">
<li><p>basecalled reads</p></li>
<li><p>alignment information</p></li>
<li><p>reference transcripts</p></li>
</ul>
</div></blockquote>
</section>
<section id="id22">
<h3>Usage example<a class="headerlink" href="#id22" title="Permalink to this headline"></a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">nanopolish</span> <span class="n">polya</span> <span class="p">[</span><span class="n">OPTIONS</span><span class="p">]</span> <span class="o">--</span><span class="n">reads</span><span class="o">=</span><span class="n">reads</span><span class="o">.</span><span class="n">fa</span> <span class="o">--</span><span class="n">bam</span><span class="o">=</span><span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">--</span><span class="n">genome</span><span class="o">=</span><span class="n">ref</span><span class="o">.</span><span class="n">fa</span>
</pre></div>
</div>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 10%" />
<col style="width: 20%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Argument name(s)</p></th>
<th class="head"><p>Required</p></th>
<th class="head"><p>Default value</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-w,</span> <span class="pre">--window=STR</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p>Compute only for reads aligning to window of reference STR (format : ctg:start_id-end_id)</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-r,</span> <span class="pre">--reads=FILE</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>the FAST(A/Q) file of native RNA reads</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-b,</span> <span class="pre">--bam=FILE</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>the BAM file of alignments between reads and the reference</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-g,</span> <span class="pre">--genome=FILE</span></code></p></td>
<td><p>Y</p></td>
<td><p>NA</p></td>
<td><p>the reference transcripts</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-t,</span> <span class="pre">--threads=NUM</span></code></p></td>
<td><p>N</p></td>
<td><p>1</p></td>
<td><p>use NUM threads</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-v,</span> <span class="pre">-vv</span></code></p></td>
<td><p>N</p></td>
<td><p>NA</p></td>
<td><p><cite>-v</cite> returns raw sample log-likelihoods, while <cite>-vv</cite> returns event durations</p></td>
</tr>
</tbody>
</table>
</section>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Wubin Ding.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>