<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Quickstart - how to estimate poly(A) tail lengths from nanopore native RNA reads &mdash; pyclustermap 1.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> pyclustermap
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">pyclustermap</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Quickstart - how to estimate poly(A) tail lengths from nanopore native RNA reads</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/quickstart_polya.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="quickstart-how-to-estimate-poly-a-tail-lengths-from-nanopore-native-rna-reads">
<span id="quickstart-polya"></span><h1>Quickstart - how to estimate poly(A) tail lengths from nanopore native RNA reads<a class="headerlink" href="#quickstart-how-to-estimate-poly-a-tail-lengths-from-nanopore-native-rna-reads" title="Permalink to this headline"></a></h1>
<p>Owing to homopolymer effects and the proximity to the sequencing adapters, the polyadenylated tails of reads obtained from nanopore native RNA sequencing are improperly basecalled, making their lengths difficult to quantify. We developed the <cite>polya</cite> subprogram to use an alternative statistical model to estimate these tail lengths.</p>
<p>In this quickstart tutorial, we’ll show you how to estimate polyadenylated tail lengths step-by-step, starting from nothing but raw fast5 files. We’ll basecall the fast5 files with Oxford Nanopore Technologies’ <em>albacore</em> basecaller, before aligning the resulting reads with <em>minimap2</em>, indexing the files with <em>nanopolish index</em>, and finally segmenting the reads and calling the tail lengths with <em>nanopolish polya</em>.</p>
<p>We’ll be following the steps taken in our <a class="reference external" href="https://github.com/paultsw/polya_analysis">benchmarking analysis</a> workflow that accompanies <a class="reference external" href="https://www.biorxiv.org/content/early/2018/11/09/459529">our publication</a>. In each step below, we’ll generate files in the working directory instead of making subdirectories, except in the case of large datasets.</p>
<section id="software-requirements">
<h2>Software requirements<a class="headerlink" href="#software-requirements" title="Permalink to this headline"></a></h2>
<ul class="simple">
<li><p><a class="reference external" href="https://github.com/jts/nanopolish">nanopolish</a> <cite>&gt;= v10.2</cite></p></li>
<li><p><a class="reference external" href="https://github.com/lh3/minimap2">minimap2</a> <cite>&gt;= v2.12</cite></p></li>
<li><p><a class="reference external" href="http://www.htslib.org/">samtools</a> <cite>&gt;= v1.9</cite></p></li>
<li><p><cite>albacore &gt;= v2.3.3</cite> (from Oxford Nanopore’s private customer portal)</p></li>
</ul>
</section>
<section id="download-raw-fast5-data-and-basecall">
<h2>Download raw fast5 data and basecall<a class="headerlink" href="#download-raw-fast5-data-and-basecall" title="Permalink to this headline"></a></h2>
<p>Let’s start by downloading a dataset of fast5 files from the European Nucleotide Archive. We’ll download a tarball of fast5 files containing reads that are known to have polyadenylated tail lengths of 30nt.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mkdir</span> <span class="n">data</span> <span class="o">&amp;&amp;</span> <span class="n">mkdir</span> <span class="n">data</span><span class="o">/</span><span class="n">fastqs</span>
<span class="n">wget</span> <span class="n">ftp</span><span class="p">:</span><span class="o">//</span><span class="n">ftp</span><span class="o">.</span><span class="n">sra</span><span class="o">.</span><span class="n">ebi</span><span class="o">.</span><span class="n">ac</span><span class="o">.</span><span class="n">uk</span><span class="o">/</span><span class="n">vol1</span><span class="o">/</span><span class="n">run</span><span class="o">/</span><span class="n">ERR276</span><span class="o">/</span><span class="n">ERR2764784</span><span class="o">/</span><span class="mi">30</span><span class="n">xpolyA</span><span class="o">.</span><span class="n">tar</span><span class="o">.</span><span class="n">gz</span> <span class="o">-</span><span class="n">O</span> <span class="mi">30</span><span class="n">xpolyA</span><span class="o">.</span><span class="n">tar</span><span class="o">.</span><span class="n">gz</span> <span class="o">&amp;&amp;</span> <span class="n">mv</span> <span class="mi">30</span><span class="n">xpolyA</span><span class="o">.</span><span class="n">tar</span><span class="o">.</span><span class="n">gz</span> <span class="n">data</span><span class="o">/</span>
<span class="n">tar</span> <span class="o">-</span><span class="n">xzf</span> <span class="n">data</span><span class="o">/</span><span class="mi">30</span><span class="n">xpolyA</span><span class="o">.</span><span class="n">tar</span><span class="o">.</span><span class="n">gz</span> <span class="o">-</span><span class="n">C</span> <span class="n">data</span><span class="o">/</span>
<span class="n">guppy_basecaller</span> <span class="o">-</span><span class="n">c</span> <span class="n">rna_r9</span><span class="mf">.4.1_70</span><span class="n">bps_hac</span><span class="o">.</span><span class="n">cfg</span> <span class="o">-</span><span class="n">i</span> <span class="n">data</span><span class="o">/</span><span class="mi">30</span><span class="n">xpolyA</span><span class="o">/</span><span class="n">fast5</span><span class="o">/</span><span class="k">pass</span> <span class="o">-</span><span class="n">s</span> <span class="n">data</span><span class="o">/</span><span class="n">fastqs</span>
<span class="n">cat</span> <span class="n">data</span><span class="o">/</span><span class="n">fastqs</span><span class="o">/*.</span><span class="n">fastq</span> <span class="o">&gt;</span> <span class="n">data</span><span class="o">/</span><span class="mi">30</span><span class="n">xpolyA</span><span class="o">.</span><span class="n">fastq</span>
</pre></div>
</div>
<p>In the above, change the value of the <cite>-f</cite> and <cite>-k</cite> arguments based on your flow-cell and sequencing kit, as the basecaller’s accuracy is highly dependent upon these settings.</p>
<p>Our directory structure should now look something like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">(</span><span class="n">current</span> <span class="n">directory</span><span class="p">)</span>
<span class="o">|-</span> <span class="n">data</span><span class="o">/</span>
<span class="o">|--</span> <span class="mi">30</span><span class="n">xpolyA</span><span class="o">.</span><span class="n">fastq</span>
<span class="o">|--</span> <span class="mi">30</span><span class="n">xpolyA</span><span class="o">.</span><span class="n">tar</span><span class="o">.</span><span class="n">gz</span>
<span class="o">|--</span> <span class="mi">30</span><span class="n">xpolyA</span><span class="o">/</span>
<span class="o">|---</span> <span class="n">fast5</span><span class="o">/</span>
<span class="o">|----</span> <span class="k">pass</span><span class="o">/</span>
<span class="o">|-----</span> <span class="n">raw_read1</span><span class="o">.</span><span class="n">fast5</span>
<span class="o">|-----</span> <span class="p">(</span><span class="o">...</span> <span class="n">more</span> <span class="n">raw</span> <span class="n">fast5</span><span class="s1">&#39;s here ...)</span>
<span class="o">|--</span> <span class="n">fastqs</span><span class="o">/</span>
<span class="o">|---</span> <span class="n">workspace</span><span class="o">/</span>
<span class="o">|----</span> <span class="k">pass</span><span class="o">/</span>
</pre></div>
</div>
</section>
<section id="index-with-nanopolish-index">
<h2>Index with nanopolish index<a class="headerlink" href="#index-with-nanopolish-index" title="Permalink to this headline"></a></h2>
<p>Let’s construct an index for our reads with nanopolish’s <cite>index</cite> subprogram. This constructs a fast lookup table that tells our program where to find the raw fast5 file for each
basecalled read.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">nanopolish</span> <span class="n">index</span> <span class="o">--</span><span class="n">directory</span><span class="o">=</span><span class="n">data</span><span class="o">/</span><span class="mi">30</span><span class="n">xpolyA</span><span class="o">/</span><span class="n">fast5</span><span class="o">/</span><span class="k">pass</span> <span class="o">--</span><span class="n">sequencing</span><span class="o">-</span><span class="n">summary</span><span class="o">=</span><span class="n">data</span><span class="o">/</span><span class="n">fastqs</span><span class="o">/</span><span class="n">sequencing_summary</span><span class="o">.</span><span class="n">txt</span> <span class="n">data</span><span class="o">/</span><span class="mi">30</span><span class="n">xpolyA</span><span class="o">.</span><span class="n">fastq</span>
</pre></div>
</div>
<p>This should generate a collection of files in the same directory that contains the <cite>30xpolyA.fastq</cite>.</p>
</section>
<section id="align-with-minimap2-and-format-the-bam-file">
<h2>Align with minimap2 and format the BAM file<a class="headerlink" href="#align-with-minimap2-and-format-the-bam-file" title="Permalink to this headline"></a></h2>
<p>Now we’ll align our basecalled mRNA sequences in the fastq file with our reference. First download a reference fasta:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="o">.</span><span class="n">githubusercontent</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">paultsw</span><span class="o">/</span><span class="n">polya_analysis</span><span class="o">/</span><span class="n">master</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">references</span><span class="o">/</span><span class="n">enolase_reference</span><span class="o">.</span><span class="n">fas</span>
<span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="o">.</span><span class="n">githubusercontent</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">paultsw</span><span class="o">/</span><span class="n">polya_analysis</span><span class="o">/</span><span class="n">master</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">references</span><span class="o">/</span><span class="n">enolase_reference</span><span class="o">.</span><span class="n">fas</span><span class="o">.</span><span class="n">fai</span>
<span class="n">mv</span> <span class="n">enolase_reference</span><span class="o">.*</span> <span class="n">data</span><span class="o">/</span>
</pre></div>
</div>
<p>Note that our directory structure should now look like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">(</span><span class="n">current</span> <span class="n">directory</span><span class="p">)</span>
<span class="o">|-</span> <span class="n">data</span><span class="o">/</span>
<span class="o">|--</span> <span class="n">enolase_reference</span><span class="o">.</span><span class="n">fas</span>
<span class="o">|--</span> <span class="n">enolase_reference</span><span class="o">.</span><span class="n">fai</span>
<span class="o">|--</span> <span class="p">(</span><span class="o">...</span> <span class="n">same</span> <span class="n">structure</span> <span class="k">as</span> <span class="n">above</span> <span class="o">...</span><span class="p">)</span>
</pre></div>
</div>
<p>Let’s run <cite>minimap2</cite> to align our basecalled reads to this reference and generate a collection of SAM/BAM files with <cite>samtools</cite>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">minimap2</span> <span class="o">-</span><span class="n">a</span> <span class="o">-</span><span class="n">x</span> <span class="nb">map</span><span class="o">-</span><span class="n">ont</span> <span class="n">data</span><span class="o">/</span><span class="n">enolase_reference</span><span class="o">.</span><span class="n">fas</span> <span class="n">data</span><span class="o">/</span><span class="mi">30</span><span class="n">xpolyA</span><span class="o">.</span><span class="n">fastq</span> <span class="o">|</span> <span class="n">samtools</span> <span class="n">view</span> <span class="o">-</span><span class="n">b</span> <span class="o">-</span> <span class="o">-</span><span class="n">o</span> <span class="n">data</span><span class="o">/</span><span class="mi">30</span><span class="n">xpolyA</span><span class="o">.</span><span class="n">bam</span>
<span class="n">cd</span> <span class="n">data</span> <span class="o">&amp;&amp;</span> <span class="n">samtools</span> <span class="n">sort</span> <span class="o">-</span><span class="n">T</span> <span class="n">tmp</span> <span class="o">-</span><span class="n">o</span> <span class="mi">30</span><span class="n">xpolyA</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> <span class="mi">30</span><span class="n">xpolyA</span><span class="o">.</span><span class="n">bam</span> <span class="o">&amp;&amp;</span> <span class="n">samtools</span> <span class="n">index</span> <span class="mi">30</span><span class="n">xpolyA</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> <span class="o">&amp;&amp;</span> <span class="n">cd</span> <span class="o">..</span>
</pre></div>
</div>
<p>Note that we used <cite>-x map-ont</cite>, which is typically for unspliced reads (e.g. genomic DNA) coming from nanopore sequencers. In typical native mRNA sequencing, you should use <cite>-x splice</cite>,
which is a splice-aware setting (and uses a different gap cost in the alignment). We’re using <cite>map-ont</cite> due to the fact that our reads come from a control dataset with no splicing.</p>
<p>There should be three more files in the <cite>data</cite> directory now: <cite>30xpolyA.bam</cite>, <cite>30xpolyA.sorted.bam</cite>, and <cite>30xpolyA.sorted.bam.bai</cite>.</p>
</section>
<section id="segmentation-and-tail-length-estimation-with-nanopolish-polya">
<h2>Segmentation and tail length estimation with nanopolish polya<a class="headerlink" href="#segmentation-and-tail-length-estimation-with-nanopolish-polya" title="Permalink to this headline"></a></h2>
<p>Finally, we can run the polyadenylation estimator:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">nanopolish</span> <span class="n">polya</span> <span class="o">--</span><span class="n">threads</span><span class="o">=</span><span class="mi">8</span> <span class="o">--</span><span class="n">reads</span><span class="o">=</span><span class="n">data</span><span class="o">/</span><span class="mi">30</span><span class="n">xpolyA</span><span class="o">.</span><span class="n">fastq</span> <span class="o">--</span><span class="n">bam</span><span class="o">=</span><span class="n">data</span><span class="o">/</span><span class="mi">30</span><span class="n">xpolyA</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> <span class="o">--</span><span class="n">genome</span><span class="o">=</span><span class="n">data</span><span class="o">/</span><span class="n">enolase_reference</span><span class="o">.</span><span class="n">fas</span> <span class="o">&gt;</span> <span class="n">polya_results</span><span class="o">.</span><span class="n">tsv</span>
</pre></div>
</div>
<p>Set the <cite>–threads</cite> flag to the number of parallel threads you want to use. Generally speaking, a larger number of threads tends to lower the compute time, but there are diminishing
returns to a higher value and performance can actually decrease if your CPU is incapable of supporting your desired number of parallel threads. The best number of threads to use is
highly dependent upon your hardware.</p>
</section>
<section id="interpreting-the-output-tsv-file">
<h2>Interpreting the output TSV file<a class="headerlink" href="#interpreting-the-output-tsv-file" title="Permalink to this headline"></a></h2>
<p>We’ll end this quickstart with a look at the output of the polya program. Let’s look at the top five lines of the <cite>polya_results.tsv</cite> file we’ve just generated:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">head</span> <span class="o">-</span><span class="mi">20</span> <span class="n">polya_results</span><span class="o">.</span><span class="n">tsv</span> <span class="o">|</span> <span class="n">column</span> <span class="o">-</span><span class="n">t</span>
<span class="n">readname</span>                              <span class="n">contig</span>   <span class="n">position</span>  <span class="n">leader_start</span>  <span class="n">adapter_start</span>  <span class="n">polya_start</span>  <span class="n">transcript_start</span>  <span class="n">read_rate</span>  <span class="n">polya_length</span>  <span class="n">qc_tag</span>
<span class="n">d6f42b79</span><span class="o">-</span><span class="mi">90</span><span class="n">c6</span><span class="o">-</span><span class="mi">4</span><span class="n">edd</span><span class="o">-</span><span class="mi">8</span><span class="n">c8f</span><span class="o">-</span><span class="mi">8</span><span class="n">a7ce0ac6ecb</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">54.0</span>          <span class="mf">3446.0</span>         <span class="mf">7216.0</span>       <span class="mf">8211.0</span>            <span class="mf">130.96</span>     <span class="mf">38.22</span>         <span class="n">PASS</span>
<span class="mi">453</span><span class="n">f3f3e</span><span class="o">-</span><span class="n">d22f</span><span class="o">-</span><span class="mi">4</span><span class="n">d9c</span><span class="o">-</span><span class="mi">81</span><span class="n">a6</span><span class="o">-</span><span class="mf">8576e23390</span><span class="n">ed</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">228.0</span>         <span class="mf">5542.0</span>         <span class="mf">10298.0</span>      <span class="mf">11046.0</span>           <span class="mf">130.96</span>     <span class="mf">27.48</span>         <span class="n">PASS</span>
<span class="n">e02d9858</span><span class="o">-</span><span class="mi">0</span><span class="n">c04</span><span class="o">-</span><span class="mi">4</span><span class="n">d86</span><span class="o">-</span><span class="mi">8</span><span class="n">dba</span><span class="o">-</span><span class="mi">18</span><span class="n">a47d9ac005</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">221.0</span>         <span class="mf">1812.0</span>         <span class="mf">7715.0</span>       <span class="mf">8775.0</span>            <span class="mf">97.16</span>      <span class="mf">29.16</span>         <span class="n">PASS</span>
<span class="n">b588dee2</span><span class="o">-</span><span class="mi">2</span><span class="n">c5b</span><span class="o">-</span><span class="mi">410</span><span class="n">c</span><span class="o">-</span><span class="mf">91e1</span><span class="o">-</span><span class="n">fe8140f4f837</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">22.0</span>          <span class="mf">8338.0</span>         <span class="mf">13432.0</span>      <span class="mf">14294.0</span>           <span class="mf">130.96</span>     <span class="mf">32.43</span>         <span class="n">PASS</span>
<span class="n">af9dfee2</span><span class="o">-</span><span class="mi">1711</span><span class="o">-</span><span class="mi">4083</span><span class="o">-</span><span class="n">b109</span><span class="o">-</span><span class="mi">487</span><span class="n">b99895e0a</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">889.0</span>         <span class="mf">3679.0</span>         <span class="mf">6140.0</span>       <span class="mf">7168.0</span>            <span class="mf">130.96</span>     <span class="mf">39.65</span>         <span class="n">PASS</span>
<span class="mi">93</span><span class="n">f98a86</span><span class="o">-</span><span class="mi">3</span><span class="n">b18</span><span class="o">-</span><span class="mi">48</span><span class="n">cf</span><span class="o">-</span><span class="mi">8</span><span class="n">c4d</span><span class="o">-</span><span class="mi">15</span><span class="n">cf277911e2</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">144.0</span>         <span class="mf">1464.0</span>         <span class="mf">5615.0</span>       <span class="mf">6515.0</span>            <span class="mf">120.48</span>     <span class="mf">30.96</span>         <span class="n">SUFFCLIP</span>
<span class="n">af9dfee2</span><span class="o">-</span><span class="mi">1711</span><span class="o">-</span><span class="mi">4083</span><span class="o">-</span><span class="n">b109</span><span class="o">-</span><span class="mi">487</span><span class="n">b99895e0a</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">889.0</span>         <span class="mf">3679.0</span>         <span class="mf">6140.0</span>       <span class="mf">7168.0</span>            <span class="mf">130.96</span>     <span class="mf">39.65</span>         <span class="n">SUFFCLIP</span>
<span class="mi">93</span><span class="n">f98a86</span><span class="o">-</span><span class="mi">3</span><span class="n">b18</span><span class="o">-</span><span class="mi">48</span><span class="n">cf</span><span class="o">-</span><span class="mi">8</span><span class="n">c4d</span><span class="o">-</span><span class="mi">15</span><span class="n">cf277911e2</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">144.0</span>         <span class="mf">1464.0</span>         <span class="mf">5615.0</span>       <span class="mf">6515.0</span>            <span class="mf">120.48</span>     <span class="mf">30.96</span>         <span class="n">PASS</span>
<span class="n">ca8d4059</span><span class="o">-</span><span class="mi">9</span><span class="n">d82</span><span class="o">-</span><span class="mi">45</span><span class="n">ee</span><span class="o">-</span><span class="n">aa07</span><span class="o">-</span><span class="mi">4</span><span class="n">b8b351618b3</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">1.0</span>           <span class="mf">2157.0</span>         <span class="mf">4255.0</span>       <span class="mf">5862.0</span>            <span class="mf">111.56</span>     <span class="mf">54.48</span>         <span class="n">PASS</span>
<span class="mi">3493</span><span class="n">c123</span><span class="o">-</span><span class="mi">78</span><span class="n">d4</span><span class="o">-</span><span class="mi">4</span><span class="n">f7c</span><span class="o">-</span><span class="n">add0</span><span class="o">-</span><span class="n">cbb249aef00a</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">78.0</span>          <span class="mf">1938.0</span>         <span class="mf">4829.0</span>       <span class="mf">5491.0</span>            <span class="mf">136.91</span>     <span class="mf">25.05</span>         <span class="n">PASS</span>
<span class="n">f5ff1802</span><span class="o">-</span><span class="mi">3</span><span class="n">fdd</span><span class="o">-</span><span class="mi">479</span><span class="n">a</span><span class="o">-</span><span class="mi">8888</span><span class="o">-</span><span class="n">c72de01bbaea</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">150.0</span>         <span class="mf">3476.0</span>         <span class="mf">7233.0</span>       <span class="mf">7932.0</span>            <span class="mf">130.96</span>     <span class="mf">25.35</span>         <span class="n">PASS</span>
<span class="n">bb929728</span><span class="o">-</span><span class="mi">2</span><span class="n">ed8</span><span class="o">-</span><span class="mi">42</span><span class="n">b0</span><span class="o">-</span><span class="n">a5a5</span><span class="o">-</span><span class="n">eea4bfd62673</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">91.0</span>          <span class="mf">1061.0</span>         <span class="mf">6241.0</span>       <span class="mf">6910.0</span>            <span class="mf">111.56</span>     <span class="mf">19.74</span>         <span class="n">PASS</span>
<span class="mi">17</span><span class="n">cf3fef</span><span class="o">-</span><span class="mi">1</span><span class="n">acb</span><span class="o">-</span><span class="mi">4045</span><span class="o">-</span><span class="mi">8252</span><span class="o">-</span><span class="n">e9c00fedfb7c</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">447.0</span>         <span class="mf">6004.0</span>         <span class="mf">20058.0</span>      <span class="mf">20964.0</span>           <span class="mf">100.40</span>     <span class="mf">25.17</span>         <span class="n">ADAPTER</span>
<span class="n">e3e38de6</span><span class="o">-</span><span class="mi">8</span><span class="n">a99</span><span class="o">-</span><span class="mi">4029</span><span class="o">-</span><span class="n">a067</span><span class="o">-</span><span class="mi">261</span><span class="n">f470517ca</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">41.0</span>          <span class="mf">1588.0</span>         <span class="mf">4057.0</span>       <span class="mf">5303.0</span>            <span class="mf">130.96</span>     <span class="mf">49.13</span>         <span class="n">PASS</span>
<span class="mi">66</span><span class="n">f55b56</span><span class="o">-</span><span class="n">c22e</span><span class="o">-</span><span class="mf">4e6</span><span class="n">d</span><span class="o">-</span><span class="mf">999e-50687</span><span class="n">bed6fb7</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">191.0</span>         <span class="mf">3160.0</span>         <span class="mf">9218.0</span>       <span class="mf">10030.0</span>           <span class="mf">125.50</span>     <span class="mf">28.79</span>         <span class="n">PASS</span>
<span class="mi">56</span><span class="n">c116d7</span><span class="o">-</span><span class="mi">9286</span><span class="o">-</span><span class="mi">4</span><span class="n">b57</span><span class="o">-</span><span class="mi">8329</span><span class="o">-</span><span class="n">e74928b11b38</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">13.0</span>          <span class="mf">1516.0</span>         <span class="mf">5845.0</span>       <span class="mf">6773.0</span>            <span class="mf">130.96</span>     <span class="mf">35.30</span>         <span class="n">PASS</span>
<span class="mi">5</span><span class="n">ca1392c</span><span class="o">-</span><span class="n">c48f</span><span class="o">-</span><span class="mi">4135</span><span class="o">-</span><span class="mi">85</span><span class="n">d3</span><span class="o">-</span><span class="mi">271</span><span class="n">bd4ee7a13</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">1.0</span>           <span class="mf">1976.0</span>         <span class="mf">4854.0</span>       <span class="mf">5947.0</span>            <span class="mf">136.91</span>     <span class="mf">44.64</span>         <span class="n">PASS</span>
<span class="mi">66</span><span class="n">b5a0ef</span><span class="o">-</span><span class="n">b8e6</span><span class="o">-</span><span class="mi">475</span><span class="n">e</span><span class="o">-</span><span class="n">bf20</span><span class="o">-</span><span class="mi">04</span><span class="n">b96154a67f</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">98.0</span>          <span class="mf">3847.0</span>         <span class="mf">7066.0</span>       <span class="mf">7925.0</span>            <span class="mf">120.48</span>     <span class="mf">29.32</span>         <span class="n">PASS</span>
<span class="mi">34</span><span class="n">bf2187</span><span class="o">-</span><span class="mi">5816</span><span class="o">-</span><span class="mi">4744</span><span class="o">-</span><span class="mf">8e6</span><span class="n">a</span><span class="o">-</span><span class="mi">3250</span><span class="n">b5247e02</span>  <span class="n">YHR174W</span>  <span class="mi">0</span>         <span class="mf">1.0</span>           <span class="mf">2897.0</span>         <span class="mf">6885.0</span>       <span class="mf">7547.0</span>            <span class="mf">125.50</span>     <span class="mf">22.54</span>         <span class="n">PASS</span>
</pre></div>
</div>
<p>Each row corresponds to the output for a given read. The columns have the following interpretation:
* <cite>readname</cite> refers to the unique ID associated to this particular read. This string is also used to look up the corresponding fast5 file, e.g. by looking</p>
<blockquote>
<div><p>through the <cite>readdb</cite> file generated by <cite>nanopolish index</cite>.</p>
</div></blockquote>
<ul>
<li><p><cite>contig</cite> refers to the reference sequence that this read aligns to, and is taken from the BAM file.</p></li>
<li><p><cite>position</cite> is the 5’ starting position of the alignment to the reference sequence, and also comes from the BAM file.</p></li>
<li><p><cite>leader_start</cite>, <cite>adapter_start</cite>, <cite>polya_start</cite>, and <cite>transcript_start</cite> are the indices of the signal segmentation generated by the underlying model within
<cite>nanopolish</cite>. Briefly, there are four biologically-meaningful regions of the raw sequence of electrical current readings within each fast5 file; these four
entries denote the starting index of each of these consecutive regions. The indices start from 0 and are oriented in the 3’-to-5’ direction (due to the
inherent orientation of the native RNA nanopore sequencing protocol). A full exposition of this segmentation algorithm is available in the
<a href="#id1"><span class="problematic" id="id2">`supplementary note&lt;https://www.biorxiv.org/content/biorxiv/suppl/2018/11/09/459529.DC1/459529-2.pdf&gt;`_</span></a> to our associated publication.</p></li>
<li><p><cite>read_rate</cite> is the estimated translocation rate (in units of nucleotides/second) of the read through the pore. The translocation rate is non-uniform during
the sequencing process of even a single molecule, so this is ultimately a summary statistic of the dynamic, time-varying rate.</p></li>
<li><p><cite>polya_length</cite> is the estimated polyadenylated tail length, in number of nucleotides. That this value is a float rather than an integer reflects the fact
that our estimated tail length is the output of an estimator based on the translocation rate.</p></li>
<li><p><cite>qc_tag</cite> is an additional flag used to indicate the validity of the estimate. Generally speaking, you should only use rows of the output file with this value
set to <cite>PASS</cite>; all other rows with (e.g.) the <cite>qc_tag</cite> set to <cite>SUFFCLIP</cite>, <cite>ADAPTER</cite>, etc. display signs of irregularity that indicate that we believe the
estimate to be unreliable. You can easily filter away all rows with the tag set to anything other than <cite>PASS</cite> using a <cite>grep</cite>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">grep</span> <span class="s1">&#39;PASS&#39;</span> <span class="n">polya_results</span><span class="o">.</span><span class="n">tsv</span> <span class="o">&gt;</span> <span class="n">polya_results</span><span class="o">.</span><span class="n">pass_only</span><span class="o">.</span><span class="n">tsv</span>
</pre></div>
</div>
</li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Wubin Ding.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>