<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Quickstart - how to align events to a reference genome &mdash; pyclustermap 1.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> pyclustermap
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">pyclustermap</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Quickstart - how to align events to a reference genome</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/quickstart_eventalign.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="quickstart-how-to-align-events-to-a-reference-genome">
<span id="quickstart-eventalign"></span><h1>Quickstart - how to align events to a reference genome<a class="headerlink" href="#quickstart-how-to-align-events-to-a-reference-genome" title="Permalink to this headline"></a></h1>
<p>The eventalign module in nanopolish is used to align events or “squiggles” to a reference genome. We (the developers of nanopolish) use this feature extensively when we want to see what the low-level signal information looks like. It helps us model the signal and discover differences in current that might hint at base modifications. Here we provide a step-by-step tutorial to help you get started with the nanopolish eventalign module.</p>
<p><strong>For more information about eventalign</strong>:</p>
<ul class="simple">
<li><p><a class="reference external" href="http://simpsonlab.github.io/2015/04/08/eventalign/">Blog post: “Aligning Nanopore Events to a Reference”</a></p></li>
<li><p><a class="reference external" href="https://www.nature.com/articles/nmeth.3444">Paper: “A complete bacterial genome assembled de novo using only nanopore sequencing data”</a></p></li>
</ul>
<p><strong>Requirements</strong>:</p>
<ul class="simple">
<li><p><a class="reference external" href="installation.html">nanopolish</a></p></li>
<li><p><a class="reference external" href="http://samtools.sourceforge.net/">samtools</a></p></li>
<li><p><a class="reference external" href="https://github.com/lh3/minimap2">minimap2</a></p></li>
</ul>
<section id="download-example-dataset">
<h2>Download example dataset<a class="headerlink" href="#download-example-dataset" title="Permalink to this headline"></a></h2>
<p>You can download the example dataset we will use here:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">wget</span> <span class="n">http</span><span class="p">:</span><span class="o">//</span><span class="n">s3</span><span class="o">.</span><span class="n">climb</span><span class="o">.</span><span class="n">ac</span><span class="o">.</span><span class="n">uk</span><span class="o">/</span><span class="n">nanopolish_tutorial</span><span class="o">/</span><span class="n">ecoli_2kb_region</span><span class="o">.</span><span class="n">tar</span><span class="o">.</span><span class="n">gz</span>
<span class="n">tar</span> <span class="o">-</span><span class="n">xvf</span> <span class="n">ecoli_2kb_region</span><span class="o">.</span><span class="n">tar</span><span class="o">.</span><span class="n">gz</span>
<span class="n">cd</span> <span class="n">ecoli_2kb_region</span>
</pre></div>
</div>
<p><strong>Details</strong>:</p>
<ul class="simple">
<li><p>Sample :    E. coli str. K-12 substr. MG1655</p></li>
<li><p>Instrument : MinION sequencing R9.4 chemistry</p></li>
<li><p>Basecaller : Albacore v2.0.1</p></li>
<li><p>Region: “tig00000001:200000-202000”</p></li>
<li><p>Note: Ligation-mediated PCR amplification performed</p></li>
</ul>
<p>This is a subset of reads that aligned to a 2kb region in the E. coli draft assembly. To see how we generated these files please refer to this section: <a class="reference internal" href="debug.html#creating-example-dataset"><span class="std std-ref">Tutorial - using extraction helper script to create example datsets</span></a>.</p>
<p>You should find the following files:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">reads.fasta</span></code> : subset of basecalled reads</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">fast5_files/</span></code> : a directory containing FAST5 files</p></li>
</ul>
<p>You will need the E. coli reference genome:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">curl</span> <span class="o">-</span><span class="n">o</span> <span class="n">ref</span><span class="o">.</span><span class="n">fa</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">ftp</span><span class="o">.</span><span class="n">ncbi</span><span class="o">.</span><span class="n">nih</span><span class="o">.</span><span class="n">gov</span><span class="o">/</span><span class="n">genomes</span><span class="o">/</span><span class="n">archive</span><span class="o">/</span><span class="n">old_genbank</span><span class="o">/</span><span class="n">Bacteria</span><span class="o">/</span><span class="n">Escherichia_coli_K_12_substr__MG1655_uid225</span><span class="o">/</span><span class="n">U00096</span><span class="o">.</span><span class="n">ffn</span>
</pre></div>
</div>
</section>
<section id="align-the-reads-with-minimap2">
<h2>Align the reads with minimap2<a class="headerlink" href="#align-the-reads-with-minimap2" title="Permalink to this headline"></a></h2>
<p>We first need to index the reads:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">nanopolish</span> <span class="n">index</span> <span class="o">-</span><span class="n">d</span> <span class="n">fast5_files</span><span class="o">/</span> <span class="n">reads</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
<p><strong>Output files</strong>: <code class="docutils literal notranslate"><span class="pre">reads.fasta.index</span></code>, <code class="docutils literal notranslate"><span class="pre">reads.fasta.index.fai</span></code>, <code class="docutils literal notranslate"><span class="pre">reads.fasta.index.gzi</span></code>, and <code class="docutils literal notranslate"><span class="pre">reads.fasta.index.readdb</span></code>.</p>
<p>Then we can align the reads to the reference:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">minimap2</span> <span class="o">-</span><span class="n">ax</span> <span class="nb">map</span><span class="o">-</span><span class="n">ont</span> <span class="o">-</span><span class="n">t</span> <span class="mi">8</span> <span class="n">ref</span><span class="o">.</span><span class="n">fa</span> <span class="n">reads</span><span class="o">.</span><span class="n">fasta</span> <span class="o">|</span> <span class="n">samtools</span> <span class="n">sort</span> <span class="o">-</span><span class="n">o</span> <span class="n">reads</span><span class="o">-</span><span class="n">ref</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> <span class="o">-</span><span class="n">T</span> <span class="n">reads</span><span class="o">.</span><span class="n">tmp</span>
<span class="n">samtools</span> <span class="n">index</span> <span class="n">reads</span><span class="o">-</span><span class="n">ref</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
<p><strong>Output files</strong>: <code class="docutils literal notranslate"><span class="pre">reads-ref.sorted.bam</span></code> and <code class="docutils literal notranslate"><span class="pre">reads-ref.sorted.bam.bai</span></code>.</p>
<p><strong>Checkpoint</strong>: Let’s see if the bam file is not truncated. This will check that the beginning of the file contains a valid header, and checks if the EOF is present. This will exit with a non-zero exit code if the conditions were not met:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">quickcheck</span> <span class="n">reads</span><span class="o">-</span><span class="n">ref</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
</section>
<section id="align-the-nanopore-events-to-a-reference">
<h2>Align the nanopore events to a reference<a class="headerlink" href="#align-the-nanopore-events-to-a-reference" title="Permalink to this headline"></a></h2>
<p>Now we are ready to run nanopolish to align the events to the reference genome:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">nanopolish</span> <span class="n">eventalign</span> \
    <span class="o">--</span><span class="n">reads</span> <span class="n">reads</span><span class="o">.</span><span class="n">fasta</span> \
    <span class="o">--</span><span class="n">bam</span> <span class="n">reads</span><span class="o">-</span><span class="n">ref</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> \
    <span class="o">--</span><span class="n">genome</span> <span class="n">ref</span><span class="o">.</span><span class="n">fa</span> \
    <span class="o">--</span><span class="n">scale</span><span class="o">-</span><span class="n">events</span> <span class="o">&gt;</span> <span class="n">reads</span><span class="o">-</span><span class="n">ref</span><span class="o">.</span><span class="n">eventalign</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
</section>
<section id="assess-the-eventalign-output">
<h2>Assess the eventalign output<a class="headerlink" href="#assess-the-eventalign-output" title="Permalink to this headline"></a></h2>
<p>If we take a peek at the first few lines of <code class="docutils literal notranslate"><span class="pre">reads-ref.eventalign.txt</span></code> this is what we get:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">contig</span>    <span class="n">position</span>  <span class="n">reference_kmer</span>  <span class="n">read_index</span>  <span class="n">strand</span>  <span class="n">event_index</span>  <span class="n">event_level_mean</span>  <span class="n">event_stdv</span>  <span class="n">event_length</span>  <span class="n">model_kmer</span>  <span class="n">model_mean</span>  <span class="n">model_stdv</span>  <span class="n">standardized_level</span>
<span class="n">gi</span><span class="o">|</span><span class="mi">545778205</span><span class="o">|</span><span class="n">gb</span><span class="o">|</span><span class="n">U00096</span><span class="mf">.3</span><span class="o">|</span><span class="p">:</span><span class="n">c514859</span><span class="o">-</span><span class="mi">514401</span>  <span class="mi">3</span>         <span class="n">ATGGAG</span>          <span class="mi">0</span>           <span class="n">t</span>       <span class="mi">16538</span>        <span class="mf">89.82</span>             <span class="mf">3.746</span>       <span class="mf">0.00100</span>       <span class="n">CTCCAT</span>      <span class="mf">92.53</span>       <span class="mf">2.49</span>        <span class="o">-</span><span class="mf">0.88</span>
<span class="n">gi</span><span class="o">|</span><span class="mi">545778205</span><span class="o">|</span><span class="n">gb</span><span class="o">|</span><span class="n">U00096</span><span class="mf">.3</span><span class="o">|</span><span class="p">:</span><span class="n">c514859</span><span class="o">-</span><span class="mi">514401</span>  <span class="mi">3</span>         <span class="n">ATGGAG</span>          <span class="mi">0</span>           <span class="n">t</span>       <span class="mi">16537</span>        <span class="mf">88.89</span>             <span class="mf">2.185</span>       <span class="mf">0.00100</span>       <span class="n">CTCCAT</span>      <span class="mf">92.53</span>       <span class="mf">2.49</span>        <span class="o">-</span><span class="mf">1.18</span>
<span class="n">gi</span><span class="o">|</span><span class="mi">545778205</span><span class="o">|</span><span class="n">gb</span><span class="o">|</span><span class="n">U00096</span><span class="mf">.3</span><span class="o">|</span><span class="p">:</span><span class="n">c514859</span><span class="o">-</span><span class="mi">514401</span>  <span class="mi">3</span>         <span class="n">ATGGAG</span>          <span class="mi">0</span>           <span class="n">t</span>       <span class="mi">16536</span>        <span class="mf">94.96</span>             <span class="mf">2.441</span>       <span class="mf">0.00125</span>       <span class="n">CTCCAT</span>      <span class="mf">92.53</span>       <span class="mf">2.49</span>        <span class="mf">0.79</span>
<span class="n">gi</span><span class="o">|</span><span class="mi">545778205</span><span class="o">|</span><span class="n">gb</span><span class="o">|</span><span class="n">U00096</span><span class="mf">.3</span><span class="o">|</span><span class="p">:</span><span class="n">c514859</span><span class="o">-</span><span class="mi">514401</span>  <span class="mi">3</span>         <span class="n">ATGGAG</span>          <span class="mi">0</span>           <span class="n">t</span>       <span class="mi">16535</span>        <span class="mf">81.63</span>             <span class="mf">2.760</span>       <span class="mf">0.00150</span>       <span class="n">NNNNNN</span>      <span class="mf">0.00</span>        <span class="mf">0.00</span>        <span class="n">inf</span>
<span class="n">gi</span><span class="o">|</span><span class="mi">545778205</span><span class="o">|</span><span class="n">gb</span><span class="o">|</span><span class="n">U00096</span><span class="mf">.3</span><span class="o">|</span><span class="p">:</span><span class="n">c514859</span><span class="o">-</span><span class="mi">514401</span>  <span class="mi">7</span>         <span class="n">AGTTAA</span>          <span class="mi">0</span>           <span class="n">t</span>       <span class="mi">16534</span>        <span class="mf">78.96</span>             <span class="mf">2.278</span>       <span class="mf">0.00075</span>       <span class="n">TTAACT</span>      <span class="mf">75.55</span>       <span class="mf">3.52</span>        <span class="mf">0.79</span>
<span class="n">gi</span><span class="o">|</span><span class="mi">545778205</span><span class="o">|</span><span class="n">gb</span><span class="o">|</span><span class="n">U00096</span><span class="mf">.3</span><span class="o">|</span><span class="p">:</span><span class="n">c514859</span><span class="o">-</span><span class="mi">514401</span>  <span class="mi">8</span>         <span class="n">GTTAAT</span>          <span class="mi">0</span>           <span class="n">t</span>       <span class="mi">16533</span>        <span class="mf">98.81</span>             <span class="mf">4.001</span>       <span class="mf">0.00100</span>       <span class="n">ATTAAC</span>      <span class="mf">95.87</span>       <span class="mf">3.30</span>        <span class="mf">0.72</span>
<span class="n">gi</span><span class="o">|</span><span class="mi">545778205</span><span class="o">|</span><span class="n">gb</span><span class="o">|</span><span class="n">U00096</span><span class="mf">.3</span><span class="o">|</span><span class="p">:</span><span class="n">c514859</span><span class="o">-</span><span class="mi">514401</span>  <span class="mi">9</span>         <span class="n">TTAATG</span>          <span class="mi">0</span>           <span class="n">t</span>       <span class="mi">16532</span>        <span class="mf">96.92</span>             <span class="mf">1.506</span>       <span class="mf">0.00150</span>       <span class="n">CATTAA</span>      <span class="mf">95.43</span>       <span class="mf">3.32</span>        <span class="mf">0.36</span>
<span class="n">gi</span><span class="o">|</span><span class="mi">545778205</span><span class="o">|</span><span class="n">gb</span><span class="o">|</span><span class="n">U00096</span><span class="mf">.3</span><span class="o">|</span><span class="p">:</span><span class="n">c514859</span><span class="o">-</span><span class="mi">514401</span>  <span class="mi">10</span>        <span class="n">TAATGG</span>          <span class="mi">0</span>           <span class="n">t</span>       <span class="mi">16531</span>        <span class="mf">70.86</span>             <span class="mf">0.402</span>       <span class="mf">0.00100</span>       <span class="n">CCATTA</span>      <span class="mf">68.99</span>       <span class="mf">3.70</span>        <span class="mf">0.41</span>
<span class="n">gi</span><span class="o">|</span><span class="mi">545778205</span><span class="o">|</span><span class="n">gb</span><span class="o">|</span><span class="n">U00096</span><span class="mf">.3</span><span class="o">|</span><span class="p">:</span><span class="n">c514859</span><span class="o">-</span><span class="mi">514401</span>  <span class="mi">11</span>        <span class="n">AATGGT</span>          <span class="mi">0</span>           <span class="n">t</span>       <span class="mi">16530</span>        <span class="mf">91.24</span>             <span class="mf">4.256</span>       <span class="mf">0.00175</span>       <span class="n">ACCATT</span>      <span class="mf">85.84</span>       <span class="mf">2.74</span>        <span class="mf">1.60</span>
</pre></div>
</div>
</section>
<section id="example-plots">
<h2>Example plots<a class="headerlink" href="#example-plots" title="Permalink to this headline"></a></h2>
<p>In <a class="reference external" href="https://www.nature.com/articles/nmeth.4184">Figure 1 of our methylation detection paper</a> we show a histogram of <code class="docutils literal notranslate"><span class="pre">event_level_mean</span></code> for a selection of k-mers to demonstrate how methylation changes the observed current. The data for these figures was generated by eventalign, which we subsequently plotted in R using ggplot2.</p>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Wubin Ding.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>